handler/oauth2/introspector_jwt_test.go

```diff
@@ -0,0 +1,95 @@
+package oauth2
+
+import (
+	"encoding/base64"
+	"strings"
+	"testing"
+
+	"github.com/ory-am/fosite"
+	"github.com/ory-am/fosite/internal"
+	"github.com/ory-am/fosite/token/jwt"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIntrospectJWT(t *testing.T) {
+	strat := &RS256JWTStrategy{
+		RS256JWTStrategy: &jwt.RS256JWTStrategy{
+			PrivateKey: internal.MustRSAKey(),
+		},
+	}
+
+	v := &JWTValidator{
+		JWTAccessTokenStrategy: strat,
+	}
+
+	for k, c := range []struct {
+		description string
+		jwt         *fosite.Request
+		tampering   func(*string)
+		expectErr   error
+	}{
+		{
+			description: "should fail because jwt is expired",
+			jwt:         jwtExpiredCase(fosite.AccessToken),
+			expectErr:   fosite.ErrRequestUnauthorized,
+		},
+		{
+			description: "should fail because signature is invalid",
+			jwt:         jwtValidCase(fosite.AccessToken),
+			tampering: func(token *string) {
+				parts := strings.Split(*token, ".")
+				dec, err := base64.StdEncoding.DecodeString(parts[1])
+				assert.NoError(t, err)
+				s := strings.Replace(string(dec), "peter", "piper", -1)
+				parts[1] = base64.StdEncoding.EncodeToString([]byte(s))
+				*token = strings.Join(parts, ".")
+			},
+			expectErr: fosite.ErrRequestUnauthorized,
+		},
+		{
+			description: "should pass",
+			jwt:         jwtValidCase(fosite.AccessToken),
+		},
+	} {
+		token, _, err := strat.GenerateAccessToken(nil, c.jwt)
+		if c.tampering != nil {
+			c.tampering(&token)
+		}
+		assert.NoError(t, err)
+		areq := fosite.NewAccessRequest(nil)
+		err = v.IntrospectToken(nil, token, fosite.AccessToken, areq, []string{})
+
+		assert.True(t, errors.Cause(err) == c.expectErr, "(%d) %s\n%s\n%s", k, c.description, err, c.expectErr)
+
+		if err == nil {
+			assert.Equal(t, areq.GetClient().GetID(), "group0")
+			assert.Equal(t, areq.Session.GetSubject(), "peter")
+		}
+
+		t.Logf("Passed test case %d", k)
+	}
+}
+
+func BenchmarkIntrospectJWT(b *testing.B) {
```

Comment:
There aren't any other benchmarks, so maybe performance is a non-goal of the library, but this was of interest to me. Understood if you want to delete in the interest of staying slim. It's certainly not useful if no one ever runs it.